river/tree/splitter/qo_splitter.py

```diff
@@ -2,26 +2,41 @@
 import math
 import typing
 
-from river.stats import Mean, Var
-from river.utils import VectorDict
+from river import stats, utils
 
 from .._attribute_test import AttributeSplitSuggestion, NumericAttributeBinaryTest
-from .attribute_observer import AttributeObserver
+from .base_splitter import Splitter
 
 
-class NumericAttributeRegressionQuantizerObserver(AttributeObserver):
-    """Quantizer observer (QO).
+class QOSplitter(Splitter):
+    """Quantization observer (QO).
 
-    Utilizes a dynamical hash-based quantization algorithm to keep track of the target statistics
-    and evaluate split candidates. This class implements the algorithm described in [^1].
-    This attribute observer keeps an internal estimator of the input feature's variance. By doing
-    that, QO can calculate better values for its radius parameter to be passed to future learning
-    nodes.
+    This splitter utilizes a hash-based quantization algorithm to keep track of the target
+    statistics and evaluate split candidates. QO, relies on the radius parameter to define
+    discretization intervals for each incoming feature. Split candidates are defined as the
+    midpoint between two consecutive hash slots.This class implements the algorithm described
+    in [^1].
+
+    The smaller the quantization radius, the more hash slots will be created to accommodate the
+    discretized data. Hence, both the running time and memory consumption increase, but the
+    resulting splits ought to be closer to the ones obtained by a batch exhaustive approach.
+    On the other hand, if the radius is too large, fewer slots will be created, less memory and
+    running time will be required, but at the cost of coarse split suggestions.
+
+    QO assumes that all features have the same range. It is always advised to scale the features
+    to apply this splitter. That can be done using the `preprocessing` module. A good "rule of
+    thumb" is to scale data using `preprocessing.StandardScaler` and define the radius as a
+    proportion of the features' standard deviation. For instance, the default radius value
+    would correspond to one quarter of the normalized features' standard deviation (since the
+    scaled data has zero mean and unit variance). If the features come from normal
+    distributions, by following the empirical rule, roughly `32` hash slots will be created.
```

Comment:
The plan is to have something like that for each splitter. Adding documentation to this one was easy because I am the author hahaha

For the other ones, any help is welcomed! @jacobmontiel, you probably know more than I do about many of the other splitters. If you want and have time for that, I would be happy to collaborate on this documentation workload.

Reply:
I improved the documentation a bit in a recent commit.